scripts/Imaging/imgpy/filters/crop_coords.py

```diff
@@ -0,0 +1,157 @@
+from __future__ import (absolute_import, division, print_function)
+import numpy as np
+import helper as h
+
+
+def cli_register(parser):
+    parser.add_argument(
+        "-R",
+        "--region-of-interest",
+        nargs='*',
+        required=False,
+        type=str,
+        help="Crop original images using these coordinates, after rotating the images.\n"
+        "If not given, the whole images are used.\n"
+        "Example: --region-of-interest='[150,234,23,22]'.")
+
+    return parser
+
+
+def gui_register(par):
+    raise NotImplementedError("GUI doesn't exist yet")
+
+
+def execute(sample, region_of_interest, flat, dark):
+    """
+    Execute the Circular Mask filter.
+
+    :param data: The sample image data as a 3D numpy.ndarray
+    :param region_of_interest: The region of interest that will be cropped
+    :param flat: The average flat image to be cropped
+    :param dark: The average dark image to be cropped
+
+    :return: the data after being processed with the filter
+    """
+    if sample is not None and flat is None and dark is not None:
+        return execute_volume(sample, region_of_interest), None, None
+    else:
+        return execute_volume(sample, region_of_interest), \
+               execute_image(flat, region_of_interest), \
+               execute_image(dark, region_of_interest)
+
+
+def _crop_coords_sanity_checks(coords, data_image, expected_data_shape=3):
+    # if nothing is provided make the user aware
+    if not isinstance(coords, list) or 4 != len(coords):
+        raise ValueError(
+            "Error in crop (region of interest) parameter (expecting a list with four integers) "
+            "[left, top, right, bottom]. Got: {0} with {1}".format(
+                coords, type(coords)))
+
+    elif not all(isinstance(crd, int) for crd in coords):
```

Comment:
I've removed the function. The length and type check for coordinates was being done early after processing the arguments, and the data sanity check is now done with `helper.check_data_stack(data)`, and I've added comments there clarifying the reasons